plaso/cli/extraction_tool.py

```diff
@@ -75,6 +75,8 @@ def _CreateProcessingConfiguration(self, knowledge_base):
     """
     # TODO: pass preferred_encoding.
     configuration = configurations.ProcessingConfiguration()
+    configuration.artifact_filters = self._artifact_filters
+    configuration.artifacts_registry = self._artifacts_registry
```

Comment:
Re: artifacts_registry

Please don't pass complex objects via configuration configuration.artifacts_registry. Configuration is meant to pass settings. If you need pass complex objects do this via the methods.

Reply:
It seems we were storing a complete ArtifactDefinitionsRegistry() in the configuration object, prior to these changes.  However, I don't think it was ever used, so I've refactored the helper to store the artifact definitions path and customer artifact definitions path only.  All tests are passing after refactoring, so I think I've accounted for any other impact.